pkg/trace/agent/log.go

```diff
@@ -144,14 +138,14 @@ func (r *ThrottledReceiver) AfterParse(args log.CustomReceiverInitArgs) error {
 	}
 
 	r.logCount = 0
-	r.tick = time.Tick(time.Duration(interval))
+	tick := time.Tick(time.Duration(interval))
 
 	// Start the goroutine resetting the log count
 	go func() {
 		defer watchdog.LogOnPanic()
 		for {
 			select {
-			case <-r.tick:
+			case <-tick:
```

Comment:
So, although the `throttledReceiver` gets the messages pushed asynchronously by seelog in its own routine, and thus not requiring a mutex to increase the log count, I think you still have a race condition here, between the log count increase (https://github.com/DataDog/datadog-agent/pull/3134/files#diff-c3f2801769d27668d4aa409b597935b9R51), and the log count reset below.

Reply:
You're right. It's quite impressive that this wasn't triggered and reported until now.

Reply:
Done!